tests/test_dse.py

```diff
@@ -168,6 +168,29 @@ def test_cse_temp_order():
     assert type(args[2]) is CTemp
 
 
+def test_cse_w_conditionals():
+    grid = Grid(shape=(10, 10, 10))
+    x, _, _ = grid.dimensions
```

Comment:
Nitpick: `x = grid.dimensions[0]` would be tidier imo

Reply:
tidier but (nitpicking) the one above performs an implicit assertion